configure.ac

```diff
@@ -348,27 +348,46 @@ OPAL_CHECK_OFI([ofi],
     [transport_ofi="yes"],
     [transport_ofi="no"])
 
-# If both OFI and portals4 requested, user needs to choose one:
-if test -n "$with_ofi" -a "$with_ofi" != "no" -a -n "$with_portals4" -a "$with_portals4" != "no" ; then
-    AC_MSG_ERROR([Cannot choose both OFI and portals4 transports, see --help for details])
-# Check which was requested, OFI or portals4:
+OMPI_CHECK_UCX([ucx],
+    [transport_ucx="yes"],
+    [transport_ucx="no"])
+
+num_transports=""
+if test -n "$with_portals4" -a "$with_portals4" != "no" ; then
+    num_transports="x$num_transports"
+fi
+if test -n "$with_ofi" -a "$with_ofi" != "no" ; then
+    num_transports="x$num_transports"
+fi
+if test -n "$with_ucx" -a "$with_ucx" != "no" ; then
+    num_transports="x$num_transports"
+fi
+
+if test -n "$num_transports" -a "$num_transports" != "x" ; then
+    AC_MSG_ERROR([Cannot select more than one transport, see --help for details])
+fi
+
+if test -n "$with_portals4" -a "$with_portals4" != "no" ; then
+    transport="portals4"
+    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
 elif test -n "$with_ofi" -a "$with_ofi" != "no" ; then
     transport="ofi"
-    transport_portals4="no"
     AC_DEFINE([USE_OFI], [1], [Define if OFI transport active])
-elif test -n "$with_portals4" -a "$with_portals4" != "no" ; then
-    transport="portals4"
-    transport_ofi="no"
-    AC_DEFINE([USE_PORTALS4], [1], [Define if Portals 4 transport active])
-# If neither, set transport to "none" and print a warning:
+elif test -n "$with_ucx" -a "$with_ucx" != "no" ; then
+    transport="ucx"
+    AC_DEFINE([USE_UCX], [1], [Define if UCX transport active])
+    AC_DEFINE([ENABLE_HARD_POLLING], [1], [Enable hard polling])
 else
     transport="none"
-    transport_ofi="no"
     transport_portals4="no"
-    AC_MSG_WARN([Neither OFI nor portals4 transport requested])
+    transport_ofi="no"
+    transport_ucx="no"
+    AC_MSG_WARN([No transport requested])
 fi
-AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
+
 AM_CONDITIONAL([USE_PORTALS4], [test "$transport_portals4" = "yes"])
+AM_CONDITIONAL([USE_OFI], [test "$transport_ofi" = "yes"])
+AM_CONDITIONAL([USE_UCX], [test "$transport_ucx" = "yes"])
 
 SANDIA_CHECK_XPMEM(
     [transport_xpmem="yes"],
```

Comment:
@jdinan Do we need to add `transport_ucx` check in line number 436 condition
```
if test "$enable_shr_atomics" != "no" -a "$transport_xpmem" = "yes" -a "$transport_ofi" = "no" -a "$transport_portals4" = "no" ;
```

Reply:
Good catch!

Reply:
Pushed a fix for this and also added shr atomics to the configury output:
```
$ ../configure --with-xpmem=$HOME/opt/xpmem --enable-pmi-simple --prefix=/home/jdinan/opt/sos
...
Compilers:
  C:              gcc
  C++:            g++
  Fortran:        gfortran

Transports:
  Portals 4:      no
  OFI:            no
  UCX:            no

On Node Communication:
  XPMEM:          yes
  CMA:            no
  memcpy (self):  yes
  Shr. atomics:   yes

Global Options:
  Remote VA:      no
  Thread support: yes
  Error checking: no

Backtrace:        execinfo gdb
```